### ISSUE TITLE
force background to 0.0 for structure factors (Trac #1157)

### ISSUE THREAD

**Issue by pkienzle** [SasView/sasview#1197]
The scale and background parameters are still present in the model in sasmodels, they are just invisible to the interface in sasview.

Unfortunately, it appears that the default value for background for structure factor is 0.001 and there is no way to change it from sasview.  Need to force it to zero within the model.

Tests should not specify background=0 in order to verify that 0 is the default value.

Note that the parameters are only hidden to sasview via sasview_model.py, they are not registered as hidden within sasmodels.  This means `sascomp hardsphere -pars` will show the defaults. May want to hide the parameters within sasmodels itself, once the default has been corrected.

Migrated from http://trac.sasview.org/ticket/1157
```json
{
    "status": "closed",
    "changetime": "2019-03-05T17:35:30",
    "_ts": "2019-03-05 17:35:30.762973+00:00",
    "description": "The scale and background parameters are still present in the model in sasmodels, they are just invisible to the interface in sasview.\n\nUnfortunately, it appears that the default value for background for structure factor is 0.001 and there is no way to change it from sasview.  Need to force it to zero within the model.\n\nTests should not specify background=0 in order to verify that 0 is the default value.\n\nNote that the parameters are only hidden to sasview via sasview_model.py, they are not registered as hidden within sasmodels.  This means `sascomp hardsphere -pars` will show the defaults. May want to hide the parameters within sasmodels itself, once the default has been corrected.\n",
    "reporter": "pkienzle",
    "cc": "",
    "resolution": "fixed",
    "workpackage": "SasView Bug Fixing",
    "time": "2018-08-17T22:19:29",
    "component": "SasView",
    "summary": "force background to 0.0 for structure factors",
    "priority": "minor",
    "keywords": "",
    "milestone": "SasView 4.3.0",
    "owner": "pkienzle",
    "type": "defect"
}
```

**Comment by pkienzle** [comment]
Trac update at `2018/09/11 16:44:41`: **pkienzle** commented:


Should background and scale should be absent from sasview for pure structure factor models?

If a user is trying to fit a structure factor directly after division of concentrated over a dilute solution measurement, having background and scale available as fit parameters may be necessary.

**Comment by pkienzle** [comment]
Trac update at `2018/09/11 21:52:28`: 
* **pkienzle** changed owner from "" to "pkienzle"
* **pkienzle** changed status from "new" to "assigned"

**Comment by butlerpd** [comment]
Trac update at `2018/09/12 00:14:46`: **butler** commented:

So question is whether to leave as two fudge factors?  My vote would be no but others might have different opinion?  If the high Q does not go to 1 it is not the S(Q). Presumably if one is pulling out a part from the whole the part is the part.  Actual experiments include background and scale but the parts should not.

**Comment by RichardHeenan** [comment]
Trac update at `2018/09/12 09:15:59`: **richardh** commented:

I would leave scale and background, as users will likely have done I(high conc)/I(low conc) but not rescaled to 1.0 at high Q.

If we want to avoid a "special case" in the interface for fitting just an S(Q) then we could have a P(Q) "none - to use with S(Q)" which is just a scalar, with ideally its background having max=min=0 by default, then fit P(Q)S(Q) in the normal way. 

Later on when we have introduced "experimental backgrounds" so users can fit the difference between two I(Q), then we could do the same for a ratio to fit 

S(Q) = scale * (Ia(Q)-BKGa)/ (Ib(Q)-BKGb) + Bkg  where Bkg is by default max=min=0

**Comment by smk78** [comment]
Trac update at `2018/09/12 09:49:12`: **smk78** commented:

I was also thinking that if you were just working with S(Q) there might be instances where, in the interests of clarity, you want to displace the curve? Having a background would allow that.

**Comment by butlerpd** [comment]
Trac update at `2018/09/12 15:58:00`: **butler** commented:

I agree with @smk78 about sometimes wanting to offset but all the use cases for that I know of have to do with graphical presentation not fitting. Thus this would be a plotting enhancement ticket and not relating to fitting.

regarding @richardh, it is true that concentration normalization may sometimes be forgotten by the user (though we can still scale a curve using the math tools so not sure it is necessary to have here).  However the use better be subtracting the background before dividing or they will get nonsense (unless of course Ia(Q)>>>background but in that case we don't need it in S(Q) either).  (Ia(Q)-BKGa)/(Ib(Q)-BKGb(Q)) is not the same as Ia(Q)/Ib(Q) - BKG!!!

That said .. not sure what @pkienzle was getting at in the question. As I understand it both scale and background remain in fact part of S(q) model but are simply suppressed at the interface.  they are still available to the scripting interface for power users.  The point of the ticket originally was to make sure the (suppressed) background was set to 0?

**Comment by RichardHeenan** [comment]
Trac update at `2018/09/12 16:13:38`: **richardh** commented:

Often the users don't know the precise concs, so having a scale is useful.

I was only looking far ahead when I suggested a fit for 

S(Q) = scale * (Ia(Q)-BKGa)/ (Ib(Q)-BKGb) + Bkg 

which is what users actually have to do, fit flat background to both data sets, and adjust scale to be their concentration ratio (and where the final Bkg is by default max=min=0, i.e. ought to be zero).

What are your thoughts on having a P(Q) = "none" to simplify the code ?

**Comment by butlerpd** [comment]
Trac update at `2018/09/12 19:10:26`: **butler** commented:

Setting P(Q) to none is an interesting idea. I'm not sure it makes the code simpler but it may (or may not) make it simpler for the new user to work with? We should consider it at any rate.  I am in fact beginning to get concerned that as we add important features we are making the interface less accessible to "non-professional" scatterers.  GUIs are such a hard thing......

**Comment by butlerpd** [comment]
Trac update at `2018/09/27 14:41:32`: 
* **butler** commented:

At this point all remaining tickets should either be blockers or critical.  Critical would be those that do not stop the actual release but need to be dealt with soon after such as updating the marketplace. Tickets such as this are blockers in that if not fixed the docs need to have an appropriate warning in them for this release.  Once that is fixed the ticket moves to 4.3 if it is not closed.



* **butler** changed priority from "major" to "blocker"

**Comment by pkienzle** [comment]
Trac update at `2018/09/28 03:57:39`: 
* **pkienzle** commented:

Apparently this is only an issue in sasmodels.  The sasview wrapper was already setting the default background to zero.


* **pkienzle** changed milestone from "SasView 4.2.0" to "SasView 4.3.0"
* **pkienzle** changed priority from "blocker" to "minor"
